projects/story-decorator/src/StoryDecorator.ts

```diff
@@ -255,6 +255,18 @@ export class StoryDecorator extends SpectrumElement {
     }
 
     protected override render(): TemplateResult {
+        // Ensure direction persistence by checking URL and localStorage
+        const urlParams = new URLSearchParams(window.location.search);
+        const urlDirection = urlParams.get('globals[textDirection]');
+
+        // If there's a direction in the URL, use it and update knobs
+        if (urlDirection === 'rtl' || urlDirection === 'ltr') {
+            this.direction = urlDirection;
+            window.__swc_hack_knobs__.defaultDirection = urlDirection;
```

Comment:
`window.__swc_hack_knobs__.defaultDirection` is a bit hacky. Can we use localstorage instead?
```js
protected override render(): TemplateResult {
    const urlParams = new URLSearchParams(window.location.search);
    const urlDirection = urlParams.get('globals[textDirection]');

    if (urlDirection && (urlDirection === 'rtl' || urlDirection === 'ltr')) {
        if (this.direction !== urlDirection) {
            this.direction = urlDirection;
            this.updateDefaultDirection(urlDirection);
        }

        if (document.documentElement.dir !== urlDirection) {
            document.documentElement.dir = urlDirection;
        }
    }
}
private updateDefaultDirection(direction: 'rtl' | 'ltr'): void {
    localStorage.setItem('defaultDirection', direction);
}
```